Assets/Scripts/NetworkManager.cs

```diff
@@ -24,15 +24,11 @@ void Start()
                 {
                     PhotonNetwork.ConnectUsingSettings(GameConstants.VERSION_STRING);
                 }
-            }
-            else
-            {
-                Object.DontDestroyOnLoad(Instance);
-                Destroy(gameObject);
+                NetworkInitialized = true;

```

Comment:
I'm not sure I follow isn't that what is already in the source?
On line 14 I check if we have set the flag and then at the end of the if statement I set the flag and there isn't an else statement